debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/processors/AuroraReselectColumnsPostProcessor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql.processors;
+
+import java.sql.SQLException;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.bean.spi.BeanRegistry;
+import io.debezium.common.annotation.Incubating;
+import io.debezium.config.Configuration;
+import io.debezium.connector.postgresql.connection.PostgresConnection;
+import io.debezium.jdbc.JdbcConfiguration;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.processors.reselect.ReselectColumnsPostProcessor;
+
+/**
+ * An extended implementation of the Debezium {@link ReselectColumnsPostProcessor}
+ * using reader instance in Aurora to reselect columns.
+ *
+ * @author Gaurav Miglani
+ */
+@Incubating
+public class AuroraReselectColumnsPostProcessor extends ReselectColumnsPostProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuroraReselectColumnsPostProcessor.class);
+
+    private static final String READER_HOST = "reader.host";
+
+    private static final String READER_PORT = "reader.port";
+
+    private String readerHost;
+
+    private int readerPort;
+
+    @Override
+    public void configure(Map<String, ?> properties) {
+        final Configuration config = Configuration.from(properties);
+        super.configure(properties);
+        this.readerHost = config.getString(READER_HOST);
+        if (this.readerHost == null || this.readerHost.isEmpty()) {
+            throw new DebeziumException("Reader host cannot be null or empty");
+        }
+
+        Integer port = config.getInteger(READER_PORT);
+        if (port == null || port <= 0) {
+            throw new DebeziumException("Reader port must be a positive integer");
+        }
+        this.readerPort = port;
+    }
+
+    @Override
+    public void close() {
+        super.close();
+        try {
+            getJdbcConnection().close();
+        }
+        catch (SQLException e) {
+            LOGGER.error("Error closing JDBC connection", e);
+        }
+    }
+
+    @Override
+    protected JdbcConnection resolveJdbcConnection(BeanRegistry beanRegistry) {
+        JdbcConnection jdbcConnection = super.resolveJdbcConnection(beanRegistry);
+        // create reader connection
+        LOGGER.info("Creating reader connection for reselect using reader host: {} and port: {}", readerHost, readerPort);
+        JdbcConfiguration newJdbcConfiguration = JdbcConfiguration.copy(jdbcConnection.config())
```

Comment:
Do we need even new config options? Maybe it would be better just to take connector level JDBC configuration and override it from options provided by the postprocessor. In that case you'd be able if necessary to override or provide other JDBC connection and conneciton string parameters.

Reply:
I like it, and that would reduce the need to create a MySQL variant too :+1:

Reply:
@jpechane `RedirectedReselectColumnsProcessor` seems good but the issue is cyclic dependency in identifying which JDBCConnection it is (PostgresConnection, MySqlConnection etc), are you saying that we shoudlwhave same class name in  all the modules ? 

For second question, are you telling to first create JDBC Configuration using connector (debezium)  jdbc config and override url, basically instead of reader.host and reader.port,  you  just want to use new key reader.jdbc.url,  along with same source (`database.user`, `database.password`)  keys ?

Reply:
@Naros  even in that case, it will be difficult to create a PostgresConnection obj/MySqlConnection obj in debezium-core package, i stuck on that part before when i try same thing in debezium-core package 🤔